src/seamless_communication/models/unit_extraction/kmeans.py

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+from torch import Tensor, nn
+import numpy as np
+from pathlib import Path
+from fairseq2.typing import Device
+from seamless_communication.assets import download_manager
+
+
+class KmeansModel(nn.Module):
+    def __init__(self, kmeans_uri: str, device: Device):
+        super().__init__()
+        km_path = download_manager.download_checkpoint(kmeans_uri, kmeans_uri)
+        km_model = np.load(km_path)
+        centroids_numpy = km_model.transpose()
+        centroids = torch.from_numpy(centroids_numpy)
+
+        self.centroids = nn.Parameter(centroids, requires_grad=False).to(device)
```

Comment:
Why do we declare this as `Parameter` if we don't require grad? Same for centroid_norm.

Reply:
will remove it